### DISCUSSION TITLE
Direct injection of LaTeXString type into a slide

### DISCUSSION THREAD

**Discussion by stefanbringuier** [ASML-Labs/PPTX.jl#27] [Ideas]
At the moment, if you want to inject mathematical expressions or equations into a `Slide` one could do the following:

```julia
using PPTX, MathJaxRenderer

pres = Presentation(;title="An equation")
slide = Slide(;title="A slide with an equation")

schrodinger_eq = Math("i\\hbar\\frac{\\partial}{\\partial t}{\\vert\\Psi(t)\\rangle} = \\hat{H}\\vert\\Psi(t)\\rangle")
eq_pic_file = tempname()*".png"
MathJaxRenderer.write(eq_pic_file,schrodinger_eq,dpi_x=300,dpi_y=300,x_zoom=10,y_zoom=10)

push!(slide,Picture(eq_pic_file,top=75,left=100,size=150))
push!(pres,slide)
PPTX.write("an_equation.pptx",pres,overwrite=true,open_ppt=false)
```

This works well, however, to me it would feel more natural to use [LaTeXStrings.jl](https://github.com/stevengj/LaTeXStrings.jl) so one doesn't need to escape all `\` and `$` symbols. Also seems more natural if one is creating notebooks making use of`LaTeXString` and/or  [Latexify](https://github.com/korsbo/Latexify.jl) and the plan is to use parts of the notebook for creating slides with `PPTX.jl`. The question would be how to inject a `LaTeXString` data type into a `Slide`. Does it get rendered by adding dependencies to `PPTX.jl`  like  (MathTexEngine.jl)[https://github.com/Kolaru/MathTeXEngine.jl) along with some plotting package and then just add as `Picture` type ? This maybe heavy. If powerpoint supports [MathML](https://w3c.github.io/mathml/)  maybe one creates a `PPTX.LaTeXEq` data type that gets properly handled upon `push!(slide,PPTX.LaTeXEq(L"\frac{1}{x}")`?

**Comment by matthijscox-asml** [comment]
For plotting, we had the idea to use [Requires.jl](https://github.com/JuliaPackaging/Requires.jl) to not force the user to install Plots as a dependency. We could try something similar for LaTeXStrings.

But I personally don't have any experience yet with Requires.jl.
